evm_arithmetization/src/recursive_verifier.rs

```diff
@@ -326,6 +327,9 @@ pub(crate) fn add_common_recursion_gates<F: RichField + Extendable<D>, const D:
     builder.add_gate_to_gate_set(GateRef::new(ExponentiationGate::new_from_config(
         &builder.config,
     )));
+    builder.add_gate_to_gate_set(GateRef::new(ConstantGate::new(
+        builder.config.num_constants,
+    )));
```

Comment:
Maybe unrelated, but if we add this to the common recursion gates, I think we may be able to remove this from the root circuit definition?

```rust
builder.add_gate(
            ConstantGate::new(inner_common_data[0].config.num_constants),
            vec![],
        );
```

Reply:
We may get trouble without it in the recursion circuits after it. Since in conditional verifying, we need a dummy circuit. But the dummy circuit will generate a constant gate while this one (root circuit) will not generate it.